### PR TITLE
[bitnami/schema-registry] fix fullName of kafka subchart 

### PR DESCRIPTION
**Pull request by Nemeq**
### Description of the change
- Changed the fullName generation on the helpers.tpl to match the name of the Kafka subchart.

### Benefits
- The chart will work with the Kafka sub chart, right now it fails because the fullName is generated incorrectly when the name override is "".

### Applicable issues

- fixes #

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Comment by rafariossaa** [comment]
Hi,
  Could you indicate how to reproduce the issue ?
  This is default for `nameOverride: ""` and I got the right name for kafa rendered for the environment variable: `SCHEMA_REGISTRY_KAFKA_BROKERS`.

**Comment by Nemeq** [comment]
Hi,
I guess is a not so common case
```
helm install kafka oci://registry-1.docker.io/bitnamicharts/schema-registry -n kafka-test-2
```
then in the schema-registry, the setting will be like this:
```
SCHEMA_REGISTRY_KAFKA_BROKERS: PLAINTEXT://kafka-kafka-controller-0.kafka-kafka-controller-headless.kafka-test-2.svc.cluster.local:9092
```
When it should be:
```
PLAINTEXT://kafka-controller-0.kafka-controller-headless.kafka-test-2.svc.cluster.local:9092
```

**Comment by rafariossaa** [comment]
The first part is the deployment name, for example, if you install it with:
```
$ helm install mysch -f values.yaml . 
```
Then I got:
```
SCHEMA_REGISTRY_KAFKA_BROKERS: "PLAINTEXT://mysch-kafka-controller-0.mysch-kafka-controller-headless.default.svc.cluster.local:9092"
```
I think that makes completely sense because that way, if you have more than one deployment of this kind in the namespace, you could differentiate them easily.

**Comment by Nemeq** [comment]
You are right about that, if we remove the default name then it will not be clear what is the actual name, the only problem is that if you do the same mistake as me and put the release as Kafka it will fail, but not sure how to prevent it.

File: bitnami/schema-registry/templates/_helpers.tpl [hunk 1]
```diff
@@ -12,8 +12,12 @@ We truncate at 63 chars because some Kubernetes name fields are limited to this
 {{- if .Values.kafka.fullnameOverride -}}
 {{- .Values.kafka.fullnameOverride | trunc 63 | trimSuffix "-" -}}
 {{- else -}}
-{{- $name := default "kafka" .Values.kafka.nameOverride -}}
+{{- if .Values.kafka.nameOverride -}}
+{{- $name := default "" .Values.kafka.nameOverride -}}
 {{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" -}}
+{{- else -}}
+{{- printf "%s" .Release.Name | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
```
[the remark this answers is not in the document]
Comment (rafariossaa) [reply bitnami/schema-registry/templates/_helpers.tpl:20]: Also, update the Chart.yaml version.

**Review by rafariossaa — CHANGES_REQUESTED** [review]

**Pull request closed without merging**